test/FingerprintApiTest.php

```diff
@@ -768,6 +770,107 @@ public function testGetRelatedVisitors429Error()
         }
     }
 
+    public function testSearchEventsWithOnlyLimit()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append(function (RequestInterface $request) {
+            $queryArray = [];
+            parse_str($request->getUri()->getQuery(), $queryArray);
+            $this->assertCount(2, $queryArray);
+            $this->assertEquals('10', $queryArray['limit']);
+
+            return $this->returnMockResponse("get_event_search_200.json");
+        });
+
+        list($events) = $this->fingerprint_api->searchEvents(10);
+
+        $this->assertCount(1, $events->getEvents());
+        $this->assertEquals("Ibk1527CUFmcnjLwIs4A9", $events->getEvents()[0]->getProducts()->getIdentification()->getData()->getVisitorId());
+    }
+
+    public function testSearchEventsWithPartialParams()
+    {
+        $this->mockHandler->reset();
+        $this->mockHandler->append(function (RequestInterface $request) {
+            $queryArray = [];
+            parse_str($request->getUri()->getQuery(), $queryArray);
+            $this->assertCount(5, $queryArray);
+            $this->assertEquals('10', $queryArray['limit']);
+            $this->assertEquals('true', $queryArray['reverse']);
+            $this->assertEquals('linked_id', $queryArray['linked_id']);
+
+            return $this->returnMockResponse("get_event_search_200.json");
+        });
+
+        list($events) = $this->fingerprint_api->searchEvents(10, self::MOCK_VISITOR_ID, null, null, "linked_id", null, null, true);
+
+        $this->assertCount(1, $events->getEvents());
+        $this->assertEquals("Ibk1527CUFmcnjLwIs4A9", $events->getEvents()[0]->getProducts()->getIdentification()->getData()->getVisitorId());
+    }
+
+    public function testSearchEventsWithAllParams()
+    {
+        $start = new DateTime('2020-01-01 00:00:00');
+        $end = new DateTime('2020-01-02 00:00:00');
+
+        $this->mockHandler->reset();
+        $this->mockHandler->append(function (RequestInterface $request) use ($end, $start) {
+            $queryArray = [];
+            parse_str($request->getUri()->getQuery(), $queryArray);
+            $this->assertCount(10, $queryArray);
+            $this->assertEquals('10', $queryArray['limit']);
+            $this->assertEquals('true', $queryArray['reverse']);
+            $this->assertEquals('linked_id', $queryArray['linked_id']);
+            $this->assertEquals($start->getTimestamp(), $queryArray['start']);
+            $this->assertEquals($end->getTimestamp(), $queryArray['end']);
+            $this->assertEquals('true', $queryArray['suspect']);
+            $this->assertEquals('good', $queryArray['bot']);
```

Comment:
Why didn't you check `bot` and `ip_adress` params?

Reply:
I've missed `ip_address`. Fixed, thanks!